client/src/components/workspace/Workspace.jsx

```diff
@@ -1,65 +1,78 @@
-import React, { useEffect, useState } from 'react';
+import React, { useState } from 'react';
 import LoadingWorkspace from './LoadingWorkspace.jsx';
 import { getWorkspaceInfo } from '../../actions/';
 
-export default ({ location: { workspaceId, neighborhood } } ) => {
-  const [space, setSpace] = useState(null);
 
-  useEffect(async () => {
-    let mnt = true;
-    if (mnt) {
-      const info = await getWorkspaceInfo(workspaceId);
-      setSpace(info);
-    }
-    return () => mnt = false;
-  },[]);
+export default ({ location: { workspaceId, neighborhood }, details = null } ) => {
+  const [space, setSpace] = useState(details);
+  // use passed in id to get details for given workspace
+  if (space === null) {
+    console.log('fetching')
+    getWorkspaceInfo(workspaceId)
+      .then(res => setSpace(res))
+      .catch(err => {
+        setSpace(false);
+        console.log(err.message);
+      })
+  }
```

Comment:
This might trigger an update with every render. I don't see another `set` call so you might be okay, but it would be safer to wrap in `useEffect`

Reply:
I believe the way  app component is structured now there shouldn't be any state changes to force a re-render since changes will happen in order from top to bottom, but I will test and reassess